base/uk.ac.stfc.isis.ibex.ui.scriptgenerator/src/uk/ac/stfc/isis/ibex/ui/scriptgenerator/views/ScriptGeneratorViewModel.java

```diff
@@ -1217,18 +1217,28 @@ public Path getScriptDefinitionsRepoPath() {
     public void reloadScriptDefinitions() {
     	scriptGeneratorModel.reloadScriptDefinitions();
     }
-    
+
+	/**
+	 * Creates prompt message for git updates available.
+	 * @return The prompt message
+	 */
+	public String getUpdatesPromptMessage() {
+		String message = "Updates to the script definitions are available. Please notify your local contact to update the script definitions at their earliest convenience.";
+		
```

Comment:
ie. 
```java
String message = "Updates to the script definitions are available. Please notify your local contact to update the script definitions at their earliest convenience.";

if (scriptGeneratorModel.isDirty()) {
 message +=  "\nThere are uncommitted changes to the script definitions. These will be lost if you update."
}

return message
```

Reply:
Yeah sorry it looks a bit funny, if I remember correctly, .isDirty wasn't able to be referenced from within scriptGeneratorModel, but only from within scriptGeneratorView? Hence still calling to get the dirtypromptmessage. It's messy but I couldn't seem to find a way to get .isDirty within the same file from where it checks for updates.

Reply:
nvm, have made changes to the logic that should address this and clean it up.